src/hotspot/share/code/codeBlob.cpp

```diff
@@ -175,12 +178,13 @@ CodeBlob::CodeBlob(const char* name, CodeBlobKind kind, int size, uint16_t heade
 {
   assert(is_aligned(size,            oopSize), "unaligned size");
   assert(is_aligned(header_size,     oopSize), "unaligned size");
+  _mutable_data = blob_end(); // Valid not null address
 }
 
 void CodeBlob::purge() {
-  if (_mutable_data != nullptr) {
+  if (_mutable_data != blob_end()) {
     os::free(_mutable_data);
-    _mutable_data = nullptr;
+    _mutable_data = blob_end(); // Valid not null address
```

Comment:
Do we still need this to be a valid address after purge(), or can we set it to nullptr here?